destination/destination_integration_test.go

```diff
@@ -0,0 +1,210 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/conduitio-labs/conduit-connector-gcp-pubsub/config"
+	"github.com/conduitio-labs/conduit-connector-gcp-pubsub/models"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/google/uuid"
+	"go.uber.org/goleak"
+	"google.golang.org/api/option"
+)
+
+func TestDestination_WriteAsync(t *testing.T) {
+	const (
+		hello    = "Hello, 世界"
+		topicFmt = "acceptance-test-topic-%s"
+
+		ignoredTopFunction = "go.opencensus.io/stats/view.(*worker).start"
+	)
+
+	var (
+		ctx     = context.Background()
+		topicID = fmt.Sprintf(topicFmt, uuid.New().String())
+	)
+
+	privateKey := os.Getenv("GCP_PUBSUB_PRIVATE_KEY")
+	if privateKey == "" {
+		t.Skip("GCP_PUBSUB_PRIVATE_KEY env var must be set")
+	}
+
+	clientEmail := os.Getenv("GCP_PUBSUB_CLIENT_EMAIL")
+	if clientEmail == "" {
+		t.Skip("GCP_PUBSUB_CLIENT_EMAIL env var must be set")
+	}
+
+	projectID := os.Getenv("GCP_PUBSUB_PROJECT_ID")
+	if projectID == "" {
+		t.Skip("GCP_PUBSUB_PROJECT_ID env var must be set")
+	}
+
+	cfg := map[string]string{
+		models.ConfigPrivateKey:  privateKey,
+		models.ConfigClientEmail: clientEmail,
+		models.ConfigProjectID:   projectID,
+		models.ConfigTopicID:     topicID,
+		models.ConfigBatchSize:   os.Getenv("GCP_PUBSUB_BATCH_SIZE"),
+		models.ConfigBatchDelay:  os.Getenv("GCP_PUBSUB_BATCH_DELAY"),
+	}
+
+	credential, err := getCredential(cfg)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if err = createTopic(ctx, projectID, topicID, credential); err != nil {
+		t.Errorf("create topic: %s", err.Error())
+	}
+
+	defer func() {
+		if err = deleteTopic(ctx, projectID, topicID, credential); err != nil {
+			t.Errorf("failed to delete topic: %s", err.Error())
+		}
+	}()
+
+	t.Run("success case", func(t *testing.T) {
```

Comment:
Shouldn't this case be covered in acceptance tests?

Reply:
Do you mean the next line?
`defer goleak.VerifyNone(t, goleak.IgnoreTopFunction(ignoredTopFunction))`

Reply:
In this test case, I just want to know that all implemented methods do not return errors:
- `Configure`
- `Open`
- `WriteAsync`
- `Flush`
- `Teardown`

But an acceptance test has a similar test case, as I understand.